server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -793,6 +818,26 @@ void handleSimpleQuery(ByteBuf buffer, final DelayableWriteChannel channel) {
         }
     }
 
+    private void handleCancelRequestBody(ByteBuf buffer, Channel channel) {
+        var keyData = readCancelRequestBody(buffer);
+        Session s = activeSessions.get(keyData); // the session executing the target query
+        if (s != null) {
+            String userName = s.sessionContext().sessionUser().name();
+            if (s.getMostRecentJobID() != null) {
+                transportKillJobsNodeAction.broadcast(
+                    new KillJobsRequest(List.of(s.getMostRecentJobID()),
```

Comment:
```suggestion
            var jobId = s.getMostRecentJobID();
            if (jobId != null) {
                transportKillJobsNodeAction.broadcast(
                    new KillJobsRequest(List.of(jobId),
```

Using a local variable instead of subsequent properties ensures that the value won't change underneath due to concurrency. Not sure if it's possible to happen here, but doesn't hurt in any case.